### PR TITLE
[Enhancement] (nereids)implement adminRebalanceDiskCommand in nereids

### LINKED ISSUE #42844: [Enhancement] (nereids)implement adminRebalanceDiskCommand in nereids
adminRebalanceDisk command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following adminRebalanceDisk command in nereids:
unsupportedadminRebalanceDiskStatement
``` ADMIN REBALANCE DISK (ON LEFT_PAREN backends+=STRING_LITERAL
        (COMMA backends+=STRING_LITERAL) RIGHT_PAREN)?                              ```

### PR DESCRIPTION
**Pull request by DongLiang-0**
### What problem does this PR solve?

Issue Number: close #42844, #42845

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 1]
```diff
@@ -4714,6 +4717,26 @@ public LogicalPlan visitShowStorageEngines(ShowStorageEnginesContext ctx) {
...
+        }
+        return new AdminRebalanceDiskCommand(false);
+    }
+
+    @Override
+    public LogicalPlan visitAdminCancelRebalanceDisk(AdminCancelRebalanceDiskContext ctx) {
+        if (ctx.ON() != null) {
+            List<String> backendList = Lists.newArrayList();
+            ctx.backends.forEach(backend -> backendList.add(stripQuotes(backend.getText())));
+            return new AdminRebalanceDiskCommand(true, backendList);
```
Comment (LiBinfeng-01) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:4735]: pls use two commands but not one command to implement this feature. Planner also use two stmt to do so

Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:4735]: As Binfeng said, you can create a base class like RebalanceDiskCommand and put common logic there, the use two different sub classes to implement the two commands

Comment (DongLiang-0) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:4735]: Thanks for pointing out the problem here, I will do.

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**